### PR TITLE
HS-213 | Refactor ClientGroupGenerationHandlera

### PR DESCRIPTION
**Pull request by pwgacek**
## Pull reviewers stats
Stats for the last 365 days:
|                                                                                                                                                                 | User               | Total reviews   | Median time to review | Total comments      |
| --------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------ | --------------- | --------------------- | ------------------- |
| <a href="https://github.com/lawos98"></a> | lawos98<br/>🥇     | **44**<br/>▀▀▀▀ | 1d 11h 30m<br/>▀▀     | **298**<br/>▀▀▀▀▀▀▀ |
| <a href="https://github.com/pwgacek"></a>                                            | pwgacek<br/>🥈     | 20<br/>▀▀       | 1d 19h 50m<br/>▀▀▀    | 88<br/>▀▀           |
| <a href="https://github.com/dudiiiiiiii"></a>                                        | dudiiiiiiii<br/>🥉 | 29<br/>▀▀       | 1d 23h 27m<br/>▀▀▀    | 18<br/>             |
| <a href="https://github.com/jagodek"></a>                                            | jagodek<br/>       | 24<br/>▀▀       | **21h 38m**<br/>▀     | 17<br/>             |

**TOTALS | PRs**: 79. **Comments:** 421. **Time to review:** 1d 8h 28m

### CODE REVIEW

File: core/src/main/java/pl/agh/edu/engine/client/ClientGroupGenerationHandler.java [hunk 1]
```diff
@@ -47,23 +45,47 @@ public List<Arrival> getArrivalsForDay(LocalTime checkInMinTime) {
...
 
-	private double getBasicNumberOfClientGroups() {
-		double popularityModifier = hotelScenariosManager.getCurrentDayMultiplier();
+	private Integer getNumberOfClientGroupsByHotelVisitPurpose(BigDecimal numberOfClientGroups,
+			HotelVisitPurpose hotelVisitPurpose) {
+		return numberOfClientGroups
+				.multiply(hotelVisitPurposeMultiplier(hotelVisitPurpose))
+				.multiply(eventMultiplier(hotelVisitPurpose))
+				.multiply(advertisementMultiplier(hotelVisitPurpose))
+				.setScale(0, RoundingMode.HALF_EVEN)
```
Comment (lawos98) [inline core/src/main/java/pl/agh/edu/engine/client/ClientGroupGenerationHandler.java:66]: statyczny import

File: core/src/main/java/pl/agh/edu/engine/client/ClientGroupGenerationHandler.java [hunk 2]
```diff
@@ -47,23 +45,47 @@ public List<Arrival> getArrivalsForDay(LocalTime checkInMinTime) {
...
+	private BigDecimal hotelVisitPurposeMultiplier(HotelVisitPurpose hotelVisitPurpose) {
+		return hotelScenariosManager.hotelVisitPurposeProbabilities.get(hotelVisitPurpose);
+	}
+
+	private BigDecimal randomMultiplier() {
+		return BigDecimal.valueOf(Math.max(0, RandomUtils.randomGaussian(1, 1. / 3)));
+	}
+
+	private BigDecimal eventMultiplier(HotelVisitPurpose hotelVisitPurpose) {
+		return BigDecimal.ONE.add(clientNumberModificationEventHandler.getCumulatedModifier().get(hotelVisitPurpose));
```
Comment (lawos98) [inline core/src/main/java/pl/agh/edu/engine/client/ClientGroupGenerationHandler.java:85]: statyczny import

File: core/src/main/java/pl/agh/edu/engine/client/ClientGroupGenerationHandler.java [hunk 3]
```diff
@@ -47,23 +45,47 @@ public List<Arrival> getArrivalsForDay(LocalTime checkInMinTime) {
...
+	}
+
+	private BigDecimal advertisementMultiplier(HotelVisitPurpose hotelVisitPurpose) {
+		return BigDecimal.ONE.add(advertisementHandler.getCumulatedModifier().get(hotelVisitPurpose));
```
Comment (lawos98) [inline core/src/main/java/pl/agh/edu/engine/client/ClientGroupGenerationHandler.java:89]: statyczny import

**Review by lawos98 — CHANGES_REQUESTED** [review]

File: core/src/main/java/pl/agh/edu/engine/client/ClientGroupGenerationHandler.java (hunk 1 above)
Comment (pwgacek) [reply core/src/main/java/pl/agh/edu/engine/client/ClientGroupGenerationHandler.java:66]: nie da się tego ustawić w linterze?

**Review by dudiiiiiiii — APPROVED** [review]

File: core/src/main/java/pl/agh/edu/engine/client/ClientGroupGenerationHandler.java (hunk 1 above)
Comment (lawos98) [reply core/src/main/java/pl/agh/edu/engine/client/ClientGroupGenerationHandler.java]: nie

**Review by lawos98 — APPROVED** [review]

**Pull request merged**